rpc/errors.go

```diff
@@ -46,205 +45,144 @@ func Err(code int, data any) *RPCError {
 // - rpcErrors: variadic list of *RPCError objects to be checked
 // Returns:
 // - error: the original error
-func tryUnwrapToRPCErr(err error, rpcErrors ...*RPCError) error {
-	var nodeErr *RPCError
-	if json.Unmarshal([]byte(err.Error()), nodeErr) != nil {
-		return err
+func tryUnwrapToRPCErr(err error, rpcErrors ...*RPCError) *RPCError {
+	nodeErr, ok := err.(*RPCError)
```

Comment:
Will this work with `ethrpc.jsonError`? Because of struct layout compatibility? In our code I've had to cast to `DataError`: `err.(ethrpc.DataError)`

Reply:
Nice catch. I've updated tryUnwrapToRPCErr so that it should work with any json error similar to RPCError (like [ethrpc.jsonErr](https://github.com/ethereum/go-ethereum/blob/master/rpc/json.go#L139))

Reply:
I would use the `Error` and `DataError` interfaces since `jsonErr` and other errors follow this interface: https://github.com/ethereum/go-ethereum/blob/0a2f33946b95989e8ce36e72a88138adceab6a23/rpc/errors.go#L36-L46